sdk/python/lib/pulumi/provider/server.py

```diff
@@ -410,6 +410,8 @@ async def Configure(self, request, context) -> proto.ConfigureResponse:
         )
 
     async def GetPluginInfo(self, request, context) -> proto.PluginInfo:
+        if self.provider.version is None:
+            raise Exception("provider version not set")
```

Comment:
I don't think this should throw. There's a lot of places we call GetInfo and will error if it returns an error. I think just returning an empty version is ok.